src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -270,11 +276,11 @@ private void DiscoverPackagesFromAllSourceBuildSources(XElement pkgSourcesElemen
                     string version = info.Version.ToLower();
 
                     // Add package with version to appropriate hashtable
-                    if (packageSource.StartsWith("source-built-"))
+                    if (packageSource.StartsWith(SourceBuiltSourceNamePrefix))
                     {
                         AddToDictionary(currentPackages, id, version);
                     }
-                    else if (packageSource.Equals("reference-packages"))
+                    else if (packageSource.Equals(ReferencePackagesSourceName))
                     {
                         AddToDictionary(referencePackages, id, version);
                     }
```

Comment:
The else case following this if else doesn't seem correct.  It assumes the package source is PSB.  If I follow correctly it, couldn't it also be the prebuilt package feed.  That is a possible source-build source according to https://github.com/dotnet/dotnet/blob/main/repo-projects/Directory.Build.targets#L141

Reply:
You are correct - will fix this.

Reply:
Fixed with https://github.com/dotnet/installer/pull/19223/commits/7993bf754a8358d5ffe51d6a4d315a852b118326